samples/graph-notification-team-channel/csharp/GraphNotification/Helper/SubscriptionManager.cs

```diff
@@ -0,0 +1,364 @@
+            {
+                if (ex.StatusCode == System.Net.HttpStatusCode.NotFound)
+                {
+                    //Subscriptions.Remove(subscription.Id);
+                    _logger.LogError(ex, $"HttpStatusCode.NotFound : Creating new subscription : {subscription.Id}");
+                    // Try and create new resource.
+
+                    await CreateSubscriptionWithResource(subscription.Resource);
+                }
+            }
+            catch (Exception ex)
+            {
+                _logger.LogError(ex, $"Update Subscription Failed: {subscription.Id}");
+            }
+        }
+        private async Task RenewSubscriptionForTeams(Subscription subscription)
+        {
+            _logger.LogWarning($"Current subscription: {subscription.Id}, Expiration: {subscription.ExpirationDateTime}");
+
+            var graphServiceClient = graphBetaClientProvider.GetGraphClientforApp();
+
+            var newSubscription = new Subscription
+            {
+                ExpirationDateTime = DateTime.UtcNow.AddHours(1)
+            };
+
+            try
+            {
+                await graphServiceClient
+                     .Subscriptions[subscription.Id]
+                     .Request()
+                     .UpdateAsync(newSubscription);
+                subscription.ExpirationDateTime = newSubscription.ExpirationDateTime;
+                _logger.LogWarning($"Renewed subscription: {subscription.Id}, New Expiration: {subscription.ExpirationDateTime}");
+            }
+            catch (Microsoft.Graph.ServiceException ex)
+            {
+                if (ex.StatusCode == System.Net.HttpStatusCode.NotFound)
+                {
+                    //Subscriptions.Remove(subscription.Id);
+                    _logger.LogError(ex, $"HttpStatusCode.NotFound : Creating new subscription : {subscription.Id}");
+                    // Try and create new resource.
+
+                    await CreateSubscriptionWithResourceForTeams(subscription.Resource);
+                }
+            }
+            catch (Exception ex)
+            {
+                _logger.LogError(ex, $"Update Subscription Failed: {subscription.Id}");
+            }
+        }
+
+        private async Task DeleteSubscription(Subscription subscription)
+        {
+            _logger.LogWarning($"Current subscription: {subscription.Id}, Expiration: {subscription.ExpirationDateTime}");
+
+            var graphServiceClient = graphBetaClientProvider.GetGraphClientforApp();
+
+            try
+            {
+                await graphServiceClient
+                     .Subscriptions[subscription.Id]
+                     .Request()
+                     .DeleteAsync();
+
+                _logger.LogWarning($"Deleted subscription: {subscription.Id}");
+            }
+            catch (Exception ex)
+            {
+                _logger.LogError(ex, $"Delete Subscription Failed: {subscription.Id}");
+            }
+        }
```

Comment:
Add proper line spacing and also add methods summary.

Reply:
Fixed